vars/stageScanForSnyk.groovy

```diff
@@ -0,0 +1,46 @@
+def call(def context, def snykAuthenticationCode, def buildFile, def organisation) {
+  if (!snykAuthenticationCode) {
+    println("Skipping Snyk Scan due to missing authentication code (to enable pass as parameter a snyk service account authentication code to this stage)")
+    return
+  }
+  if (!organisation) {
+    organisation = context.projectId
+    println("organisation set to ${organisation}")
+  }
+
+  String message = "Snyk scan mode: build will " + (context.failOnSnykScanVulnerabilities ? "" : "not ") +
+  "fail if vulnerabilities are found (failOnSnykScanVulnerabilities=${context.failOnSnykScanVulnerabilities})!"
+  println(message)
+
+  if (buildFile == null) {
+    error "build file definition for snyk security scan is null!"
+  } else {
+    stage('Snyk Security Scan') {
+      withEnv(["SNYK_AUTHENTICATION_CODE=${snykAuthenticationCode}", "PROJECT_NAME=${context.targetProject}",
+               "COMPONENT_NAME=${context.componentId}", "BUILD_FILE=${buildFile}", "ORGANISATION=${organisation}"]) {
+        // Verify that snyk is installed
+        def status = sh(script: "snyk version", returnStatus: true)
+        if (status != 0) {
+          error "snyk is not installed!"
+        }
+        // Authorise snyk
+        status = sh(script: "snyk auth $SNYK_AUTHENTICATION_CODE", returnStatus: true)
+        if (status != 0) {
+          error "something went wrong by authorising snyk (SNYK_AUTHENTICATION_CODE=$SNYK_AUTHENTICATION_CODE)!"
+        }
+        // first monitor project
+        status = sh(script: "snyk monitor --org=$ORGANISATION --file=$BUILD_FILE --project-name=$COMPONENT_NAME", returnStatus: true)
+        if (status != 0) {
+          error "something went wrong with snyk monitor command!"
```

Comment:
Would it be possible to capture stderr or so? Otherwise the error message is quite cryptic ...

Reply:
I did many test based on the sh command docs: https://jenkins.io/doc/pipeline/steps/workflow-durable-task-step/#code-sh-code-shell-script but I couldn't capture stderr or stdout.

Reply:
pls see comment related to question about "status" handling snyk auth call